### ISSUE TITLE
in nmt_with_attention, the gru in decoder is not connected, last step state is not passed to this step

### ISSUE THREAD

**Issue by hihell** [tensorflow/tensorflow#38248]
https://www.tensorflow.org/tutorials/text/nmt_with_attention

the decoder is trained step by step, and it's not passing last step state to this step

```
# passing the concatenated vector to the GRU
output, state = self.gru(x)
```

is this a feature or a bug? I checked a lot of NMT with attention paper, unlike the document those decoder are connected.

thanks in advance!

**Comment by ManishAradwad** [comment]
Hello, I'd like to work on this issue. Can you plz give me some pointers on where I should start?

**Comment by MarkDaoust** [comment]
I think you're right.

@ManishAradwad if you want to take a shot at a fix, I think all it needs is a `output, state = self.gru(x, initial_state=hidden)`

**Comment by yashk2810** [comment]
Be careful with that. The last time I did that, the attention plots didn't come out right. So please test and see if the attention plots remain similar to what they are now.

**Comment by hihell** [comment]
@MarkDaoust  
I agree with @yashk2810
I did exact the same with `initial_state=hidden` and the attention is strange, and the train loss is not converging after 100 epochs (it goes down at frist then goes up, ended at 0.2)

**Comment by MarkDaoust** [comment]
Thanks for taking a look @hihell  

> the attention is strange

Can you be more specific?

I'll dive in and try to figure out what's wrong.

**Comment by ManishAradwad** [comment]
I tried re-running the notebook with `initial_state=hidden`, the loss after 10 epochs is 0.076(Previously it was 0.0958). The attention plots are not exactly same but the translations are more or less similar.  
For ex. Previously `try to find out` is now `try to figure it out`
Are these significant differences?

**Comment by hihell** [comment]
@MarkDaoust 
![Figure_1](https://user-images.githubusercontent.com/1401509/78970283-8998c980-7b3b-11ea-990a-98779e96ba84.png)
![Figure_2](https://user-images.githubusercontent.com/1401509/78970331-a6cd9800-7b3b-11ea-9aab-fcc9b2802afb.png)
this result is trained by 60k samples in 10 epochs

**Comment by hihell** [comment]
@MarkDaoust 
hi, this there any update?

**Comment by MarkDaoust** [comment]
I'm working on this now.

**Comment by nikitamaia** [comment]
Fixed in [9e18593](https://github.com/tensorflow/docs/commit/9e18593b7ee19058b651626d0e96a552ac7b4733)
